### PR TITLE
fix Client.destroy bugs

### PR DESCRIPTION
**Pull request by isonmad**
There were a bunch of them. the ClientManager promise bug hid the fact that the logout() call was always failing, and the logout call failing made sure the faulty timeout-clearing code in Client.destroy never got called.

### CODE REVIEW

**Comment by iCrawl** [comment]
I have not fully read through it yet, but you should change the branch from `master` to `indev`.

**Comment by GusCaplan** [comment]
AHH

you aren't supposed to use the client after you call destroy

**Comment by Gawdl3y** [comment]
I'm not sure what this is even attempting to fix. It might be more clear if you rebase your modifications onto the current indev branch.

**Comment by Gawdl3y** [comment]
Ah, I see the overwriting of `_timeouts` and `_intervals` to arrays - that was left over from when they actually were arrays.
Not sure what the issue you're having with logout is, though.
That promise simplification is also good.

File: src/client/Client.js [hunk 1]
```diff
@@ -225,17 +225,14 @@ class Client extends EventEmitter {
...
-        for (const t of this._timeouts) clearTimeout(t);
-        for (const i of this._intervals) clearInterval(i);
-        this._timeouts = [];
-        this._intervals = [];
-        this.token = null;
-        this.email = null;
-        this.password = null;
-        resolve();
-      }).catch(reject);
+    return this.manager.destroy().then(() => {
```
Comment (Gawdl3y) [inline src/client/Client.js]: This simplification is good. :+1:

File: src/client/ClientManager.js [hunk 2]
```diff
@@ -57,14 +57,11 @@ class ClientManager {
...
   destroy() {
-    return new Promise((resolve) => {
-      if (!this.client.user.bot) {
-        this.client.rest.methods.logout().then(resolve);
-      } else {
-        this.client.ws.destroy();
-        resolve();
-      }
-    });
+    let p = Promise.resolve();
```
Comment (Gawdl3y) [inline src/client/ClientManager.js]: This is creating an unnecessary promise in the case it isn't a bot account.

File: src/client/rest/RESTMethods.js [hunk 3]
```diff
@@ -33,7 +33,7 @@ class RESTMethods {
   }
 
   logout() {
-    return this.rest.makeRequest('post', Constants.Endpoints.logout, true);
+    return this.rest.makeRequest('post', Constants.Endpoints.logout, true, {});
```
Comment (Gawdl3y) [inline src/client/rest/RESTMethods.js]: I don't think this is necessary.

Comment (isonmad) [inline src/client/rest/RESTMethods.js]: Without this change logout fails. What's the alternative?

**Pull request merged**